lib/rex/socket.rb

```diff
@@ -193,10 +193,10 @@ def self.getaddresses(hostname, accept_ipv6 = true)
       return [hostname]
     end
 
-    res = if @@resolver
-      self.rex_getaddrinfo(hostname)
+    if @@resolver
+      res = self.rex_getaddrinfo(hostname)
     else
-      ::Addrinfo.getaddrinfo(hostname, 0, ::Socket::AF_UNSPEC, ::Socket::SOCK_STREAM)
+      res = ::Addrinfo.getaddrinfo(hostname, 0, ::Socket::AF_UNSPEC, ::Socket::SOCK_STREAM)
```

Comment:
Sorry, Rust-y patterns used to get something owned by context outside the conditional leaking into my Ruby code. Appreciate the fix.